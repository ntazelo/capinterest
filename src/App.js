import { Route, Routes } from 'react-router-dom';
import Header from './Components/Header/Header';
import Categories from './Components/Categories/Categories';
import CategoryDetails from './Components/CategoryDetails/CategoryDetails';
import './App.css';

function App() {
  return (
    <div>
      <Header />
      <Routes>
        <Route path="/" element={<Categories />} />
        <Route path="/:name" element={<CategoryDetails />} />
      </Routes>
    </div>
  );
}

export default App;
