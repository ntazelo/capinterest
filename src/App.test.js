import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from './Redux/ConfigureStore';
import App from './App';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing the app', () => {
  it('testing HomePage UI', () => {
    act(() => {
      render(
        <React.StrictMode>
          <BrowserRouter>
            <Provider store={store}>
              <App />
            </Provider>
          </BrowserRouter>
        </React.StrictMode>, container,
      );
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
