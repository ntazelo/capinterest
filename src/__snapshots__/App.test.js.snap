// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app testing HomePage UI 1`] = `
"<div>
  <header class=\\"navigation-cont\\"><a class=\\"back-link-cont\\" href=\\"/\\"><i class=\\"fas fa-chevron-left\\"></i><span class=\\"link-hint\\">Menu</span></a><span class=\\"on-page-hint\\">Category</span>
    <nav class=\\"icons-cont\\">
      <ul class=\\"icons-list-wrapper\\">
        <li class=\\"icon\\"><i class=\\"fas fa-microphone\\"></i></li>
        <li class=\\"icon\\"><i class=\\"fas fa-cog\\"></i></li>
      </ul>
    </nav>
  </header>
  <section class=\\"intro-cont\\">
    <div class=\\"img-gal\\"></div>
    <div class=\\"intro-details\\">
      <div class=\\"content-wrapper\\">
        <h1 class=\\"title-cont\\">Capinterest</h1>
        <p class=\\"category-count\\">6 Category</p>
      </div>
    </div>
  </section>
  <section class=\\"filter-cont\\">
    <p class=\\"indic-sect\\">filter by category</p>
    <form class=\\"form-control\\"><select name=\\"category\\" class=\\"category-input\\" required=\\"\\">
        <option value=\\"category\\">Category</option>
        <option value=\\"sports\\">sports</option>
        <option value=\\"computer\\">computer</option>
        <option value=\\"music\\">music</option>
        <option value=\\"food\\">food</option>
        <option value=\\"animals\\">animals</option>
        <option value=\\"fashion\\">fashion</option>
      </select><button type=\\"submit\\" class=\\"btn-book\\">Add book</button></form>
  </section>
  <section class=\\"categories-cont\\"></section>
</div>"
`;
